frame/system/src/lib.rs

```diff
@@ -133,7 +133,7 @@ pub fn extrinsics_root<H: Hash, E: codec::Encode>(extrinsics: &[E]) -> H::Output
 
 /// Compute the trie root of a list of extrinsics.
 pub fn extrinsics_data_root<H: Hash>(xts: Vec<Vec<u8>>) -> H::Output {
-	H::ordered_trie_root(xts)
+	H::ordered_trie_root(xts, sp_core::StateVersion::V0)
```

Comment:
> Looks already much better, however I think we need to get rid off the feature. The problem with the feature is cargo and that it activates it everywhere once you have activated it somewhere. The problem I'm seeing are parachain projects that depend on polkadot and thus would pull in the feature activation into their native runtime the moment we activate this in Polkadot. This is bad.
> 
> So, I would say we make the following change:
> 
>     1. Drop the feature.
> 
>     2. Bump the `Core` version to 4 as you already have done it.
> 
>     3. Add the `state_version` field to the `RuntimeVersion` and make sure we decode the new `RuntimeVersion` when `Core >= 4`. (Maybe while doing this, we should make `RuntimeVersion` directly versionable and put it into some `enum` aka `enum RuntimeVersion { V0(), V1(), V2 }`.
> 
>     4. Then everything can refer to the `state_version` in the runtime wasm to get it. frame system would also take it from there. This would then mean that we will just need to bump the `state_version` in the runtime version when we are ready.

Well actually the feature should only be use for runtime building, but I know how it can be tricky to do sometime.

I previously did add a state_version to the runtime version but revert as I did not want the overhead and thought at the time that it could still be done at the time sp_core need an actual update (would use 1 additional core version though).
But if it is fine to have this in all headers, good (will still be doable to change runtime version format backward when this is not of any use).
Aggree, we should `enum RuntimeVersion` (instead of using an `OldOldRuntimeVersion`).
(will probably update to this on monday)

Reply:
> But if it is fine to have this in all headers

What you mean by this? Having what in the headers?

Reply:
yes, not header, just wasm

Reply:
Yeah that should be fine. I don't see any problem.